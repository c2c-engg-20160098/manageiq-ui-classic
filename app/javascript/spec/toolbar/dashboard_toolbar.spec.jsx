import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import DashboardToolbar from '../../components/dashboard_toolbar';

const dashboardProps = {
  allowAdd: true,
  allowReset: true,
  locked: false,
  items: [
    {
      id: 31,
      type: "button",
      text: "add",
      image: "fa fa-pie-chart fa-lg",
      title: "Add this Chart Widget"
    },
  ],
}

describe('<DashboardToolbar />', () => {
  it('renders ok', () => {
    const t = mount(<DashboardToolbar {...dashboardProps} />);
    expect(toJson(t)).toMatchSnapshot();
  });
});
