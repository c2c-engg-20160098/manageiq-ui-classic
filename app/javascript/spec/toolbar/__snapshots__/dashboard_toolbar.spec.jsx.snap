// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DashboardToolbar /> renders ok 1`] = `
<DashboardToolbar
  allowAdd={true}
  allowReset={true}
  items={
    Array [
      Object {
        "id": 31,
        "image": "fa fa-pie-chart fa-lg",
        "text": "add",
        "title": "Add this Chart Widget",
        "type": "button",
      },
    ]
  }
  locked={false}
>
  <div
    className="toolbar-pf-actions miq-toolbar-actions"
  >
    <div
      className="miq-toolbar-group form-group"
    >
      <Uncontrolled(Dropdown)
        disabled={false}
        id="dropdown-custom-2"
      >
        <Dropdown
          bsClass="dropdown"
          componentClass={[Function]}
          disabled={false}
          id="dropdown-custom-2"
          onToggle={[Function]}
        >
          <ButtonGroup
            block={false}
            bsClass="btn-group"
            className="dropdown"
            justified={false}
            vertical={false}
          >
            <div
              className="dropdown btn-group"
            >
              <DropdownToggle
                bsClass="dropdown-toggle"
                bsRole="toggle"
                disabled={false}
                id="dropdown-custom-2"
                key=".0"
                onClick={[Function]}
                onKeyDown={[Function]}
                open={false}
                title="Add a widget"
                useAnchor={false}
              >
                <Button
                  active={false}
                  aria-expanded={false}
                  aria-haspopup={true}
                  block={false}
                  bsClass="btn"
                  bsStyle="default"
                  className="dropdown-toggle"
                  disabled={false}
                  id="dropdown-custom-2"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="button"
                  title="Add a widget"
                >
                  <button
                    aria-expanded={false}
                    aria-haspopup={true}
                    className="dropdown-toggle btn btn-default"
                    disabled={false}
                    id="dropdown-custom-2"
                    onClick={[Function]}
                    onKeyDown={[Function]}
                    role="button"
                    title="Add a widget"
                    type="button"
                  >
                    <span
                      className="fa fa-plus fa-lg"
                    />
                     
                    <span
                      className="caret"
                    />
                  </button>
                </Button>
              </DropdownToggle>
              <DropdownMenu
                bsClass="dropdown-menu"
                bsRole="menu"
                className="scrollable-menu"
                key=".1"
                labelledBy="dropdown-custom-2"
                onClose={[Function]}
                onSelect={[Function]}
                pullRight={false}
              >
                <RootCloseWrapper
                  disabled={true}
                  event="click"
                  onRootClose={[Function]}
                >
                  <ul
                    aria-labelledby="dropdown-custom-2"
                    className="scrollable-menu dropdown-menu"
                    role="menu"
                  >
                    <MenuItem
                      bsClass="dropdown"
                      disabled={false}
                      divider={false}
                      header={false}
                      key=".$31"
                      onClick={[Function]}
                      onKeyDown={[Function]}
                      onSelect={[Function]}
                    >
                      <li
                        className=""
                        role="presentation"
                      >
                        <SafeAnchor
                          componentClass="a"
                          onClick={[Function]}
                          onKeyDown={[Function]}
                          role="menuitem"
                          tabIndex="-1"
                        >
                          <a
                            href="#"
                            onClick={[Function]}
                            onKeyDown={[Function]}
                            role="menuitem"
                            tabIndex="-1"
                          >
                            <i
                              className="fa fa-pie-chart fa-lg"
                            />
                            Â 
                            add
                          </a>
                        </SafeAnchor>
                      </li>
                    </MenuItem>
                  </ul>
                </RootCloseWrapper>
              </DropdownMenu>
            </div>
          </ButtonGroup>
        </Dropdown>
      </Uncontrolled(Dropdown)>
      <button
        className="btn btn-default"
        onClick={[Function]}
        title="Reset Dashboard Widgets to the defaults"
        type="button"
      >
        <i
          className="fa fa-reply fa-lg"
        />
      </button>
    </div>
  </div>
</DashboardToolbar>
`;
